compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2460,6 +2471,13 @@ private boolean hasSameReadonlyFlag(BType source, BType target) {
         }
 
         public Boolean visit(BTupleType t, BType s) {
+            if (((!t.tupleTypes.isEmpty() && t.tupleTypes.stream()
+                    .allMatch(mem -> checkTupleMemberIsNoType(mem))) ||
```

Comment:
Shall we replace the stream here too?  I think we can change `checkTupleMemberIsNoType` to accept `t.tupleTypes` instead?